### PR TITLE
Schema updated

### PR DESCRIPTION
**Pull request by mdhishaamakhtar**

### CODE REVIEW

File: model/Action.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    default: []
+  },
+  correct: {
+    type: String,
+    required: false
+  }
+});
+const actionSchema = new mongoose.Schema({
+  timestamp: {
+    type: Number,
```
Comment (L04DB4L4NC3R) [inline model/Action.js]: Type should be `Date`

File: routes/addAction.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//jshint esversion:8
+const router = require("express").Router();
+const Event = require("../model/Event");
+const Action = require("../model/Action");
+const verify = require("./verifyToken");
+
+router.post("/:event_id", verify, (req, res) => {
+  const eventID = req.params.event_id;
+  const action = new Action({
+    action_type: req.body.action_type
```
Comment (L04DB4L4NC3R) [inline routes/addAction.js]: Validate action_type

File: routes/addOption.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+router.post("/:ActionID/:QuestionID", verify, (req, res) => {
+  const ActionID = req.params.ActionID;
+  const QuestionID = req.params.QuestionID;
+  Action.updateOne(
+    { _id: ActionID, "Questions._id": QuestionID },
+    { $push: { "Questions.$.options": req.body } },
+    err => {
+      if (err) {
+        res.send(err);
```
Comment (L04DB4L4NC3R) [inline routes/addOption.js]: Send JSON always

File: routes/addOption.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      } else {
+        res.send("Successful");
```
Comment (L04DB4L4NC3R) [inline routes/addOption.js]: here too

File: routes/addQuestion.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  Action.findByIdAndUpdate(
+    ActionID,
+    { $push: { Questions: req.body } },
+    err => {
+      if (err) {
+        res.json(err);
+      }
+    }
+  );
+  await Action.findById(ActionID)
```
Comment (L04DB4L4NC3R) [inline routes/addQuestion.js]: This func will execute before the previous one completes

File: routes/auth.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  //Create a Token
+  const token = jwt.sign(
+    {
+      _id: user._id
+    },
+    process.env.TOKEN_SECRET
+  );
+  res.header("auth-token", token).send(token);
```
Comment (L04DB4L4NC3R) [inline routes/auth.js]: Use JSON

File: validation.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  return schema.validate(data);
+};
+const loginValidation = data => {
+  const schema = Joi.object({
+    email: Joi.string()
+      .min(6)
+      .required()
+      .email(),
+    password: Joi.string()
+      .min(6)
```
Comment (L04DB4L4NC3R) [inline validation.js]: In DB min 8 but here min 6?

File: routes/auth.js (hunk 6 above)
Comment (mdhishaamakhtar) [reply routes/auth.js]: Resolved it.

File: validation.js (hunk 7 above)
Comment (mdhishaamakhtar) [reply validation.js]: Resolved it.

File: model/Action.js (hunk 1 above)
Comment (mdhishaamakhtar) [reply model/Action.js]: Resolved it

File: routes/addAction.js (hunk 2 above)
Comment (mdhishaamakhtar) [reply routes/addAction.js]: Resolved it

File: routes/addOption.js (hunk 3 above)
Comment (mdhishaamakhtar) [reply routes/addOption.js]: Resolved it

File: routes/addOption.js (hunk 4 above)
Comment (mdhishaamakhtar) [reply routes/addOption.js]: Resolved it

File: routes/addQuestion.js (hunk 5 above)
Comment (mdhishaamakhtar) [reply routes/addQuestion.js]: Working on it

File: index.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+const addQuestion = require("./routes/addQuestion");
+const addOption = require("./routes/addOption");
+const nextQuestion = require("./routes/nextQuestion");
+const updateStat = require("./routes/updateStat");
+const publishQuestion = require("./routes/publishQuestion");
+const chooseOption = require("./routes/chooseOption");
+const getEventdetail = require("./routes/getEventdetail");
+const getActiondetail = require("./routes/getActiondetail");
+app.use("/api/user", authRoute);
+app.use("/api/addEvent", addEvent);
```
Comment (L04DB4L4NC3R) [inline index.js]: Instead of routes like `/addEvent` just have `/event`

File: index.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+app.use("/api/addAction", addAction);
+app.use("/api/addQuestion", addQuestion);
+app.use("/api/addOption", addOption);
+app.use("/api/nextQuestion", nextQuestion);
+app.use("/api/updateStat", updateStat);
+app.use("/api/publishQuestion", publishQuestion);
+app.use("/api/chooseOption", chooseOption);
+app.use("/api/getEventdetail", getEventdetail);
```
Comment (L04DB4L4NC3R) [inline index.js]: This can be included in the `/event` route. Group all event related endpoints here

File: index.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+const getEventdetail = require("./routes/getEventdetail");
+const getActiondetail = require("./routes/getActiondetail");
+app.use("/api/user", authRoute);
+app.use("/api/addEvent", addEvent);
+app.use("/api/addAction", addAction);
+app.use("/api/addQuestion", addQuestion);
+app.use("/api/addOption", addOption);
+app.use("/api/nextQuestion", nextQuestion);
+app.use("/api/updateStat", updateStat);
+app.use("/api/publishQuestion", publishQuestion);
```
Comment (L04DB4L4NC3R) [inline index.js]: Group all question related endpoints in `/questions` route

File: package.json [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    "@hapi/joi": "^17.1.0",
+    "bcrypt-rust-wasm": "^0.1.8",
+    "body-parser": "^1.19.0",
+    "cors": "^2.8.5",
+    "dotenv": "^8.2.0",
+    "express": "^4.17.1",
+    "jsonwebtoken": "^8.5.1",
+    "locks": "^0.2.2",
+    "mongoose": "^5.9.0",
+    "nodemon": "^2.0.2",
```
Comment (L04DB4L4NC3R) [inline package.json]: This should be a dev dependency

File: routes/getActiondetail.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
+//jshint esversion:6
+const router = require("express").Router();
+const Action = require("../model/Action");
+
+router.get("/:ActionID", (req, res) => {
+  const ActionID = req.params.ActionID;
```
Comment (L04DB4L4NC3R) [inline routes/getActiondetail.js]: Use camelcase for variables

Comment (L04DB4L4NC3R) [reply routes/getActiondetail.js]: Do this for all such variables

File: index.js (hunk 8 above)
Comment (mdhishaamakhtar) [reply index.js]: Done

File: index.js (hunk 9 above)
Comment (mdhishaamakhtar) [reply index.js]: Done

File: index.js (hunk 10 above)
Comment (mdhishaamakhtar) [reply index.js]: Done

File: package.json (hunk 11 above)
Comment (mdhishaamakhtar) [reply package.json]: What does a dev dependency

File: routes/getActiondetail.js (hunk 12 above)
Comment (mdhishaamakhtar) [reply routes/getActiondetail.js]: Done

**Comment by Geek-ubaid** [comment]
@L04DB4L4NC3R  see the requested changes are done and merge this PR.

**Comment by mdhishaamakhtar** [comment]
All requested changes have been made.

**Comment by L04DB4L4NC3R** [comment]
> All requested changes have been made.

Approved

**Pull request merged**